### ISSUE TITLE
tv will not switch inputs

### ISSUE THREAD

**Issue by reaps101** [JPersson77/LGTVCompanion#227]
### Describe the bug

Tv will power on and off fine but if tv is on say hdmi 2, when powering on it will not switch to the pc's hdmi 4

### How to reproduce the bug?

leave tv on hdmi 2, turn tv off, start pc, tv turns on but does not switch to pc input

### What was the expected behavior

tv switch to hdmi 4

### What is your system?

Win 11, build 22631.3737, LG OLED C1

### Log file and screenshots

[https://pastebin.com/1WH7wHu4](url)

**Comment by JPersson77** [comment]
Hi, can you double check the pastebin link please. It does not seem to work for me. Cheers

**Comment by reaps101** [comment]
seems to have fixed itself, it didnt work after i updated to the latest version, but after switching the settings off and on a couple of times it seems to have fixed it

**Comment by JPersson77** [comment]
The only thing that stood out in the log was that the TV was quite slow to respond. Not sure what to make of it but maybe it is related

**Comment by reaps101** [comment]
You got me wondering now if a factory reset of the tv would be a good idea

**Comment by JPersson77** [comment]
Yeah it wouldnt be the first time it helped with weird issues. Maybe start with pulling the plug for a few minutes though and see if things run smoothly from there

**Comment by hapklaar** [comment]
Hmm, interesting. I have the same or similar issue. Instead of creating a new issue I thought I'd add here.

I have two PC's connected to an LG C2, one on HDMI1 and the other on HDMI4. The one connected to HDMI1 always works correctly and switches the input to HDMI1 when turned on. 

The laptop connected to HDMI4 never switches to HDMI4 when turned on.

I pasted a debug log here: https://pastebin.com/m430VVtt

**Comment by JPersson77** [comment]
Hi @hapklaar there are no issues with your log, everything looks as expected. Since the TV powers on fine and responds to the request to switch HDMI, but doesn't actually do it, Isuspect that the TV needs slightly more time after being powered on before the HDMI-switch command is sent. Currently on vacation so it'll take a few weeks but I'll have a look

**Comment by hapklaar** [comment]
Thanks @JPersson77 for responding while on holiday! No rush at all, I hope you have a good one!

In the mean time, I'll do some additional testing.

**Comment by hapklaar** [comment]
I've been testing with alternate inputs, but all behave the same as hdmi4. It works only 1 out of 20something times. Your suspicion about it being a timing issues might be correct, but then I would expect input switching to occur always when the TV is already on but on the wrong input. This doesn't seem to be the case, or maybe I tested it wrong...

**Comment by vladimir-fsh** [comment]
I have a similar issue, if a last input was `AppleTV` then enabling `PC` turns on `TV` + `AppleTV` on its source. I waited quite long time and nothing happened. I have to switch a source manually to `PC`.

**Comment by JPersson77** [comment]
Thanks for reporting! I will have a look as soon as possible. Just for completeness, can you please enable debug level logging and paste a link here wirh the full log

**Comment by vladimir-fsh** [comment]
@JPersson77 Here it is. 

My steps are:
1. Turn on PC (hdmi 1), AppleTv is sleeping. TV was turned on by PC
2. Sleep my PC, TV was turned off by PC
3. Turn on AppleTV. TV was turned on by AppleTV. Turned off Apple TV, TV was turned off by AppleTV
4. Turn on PC. TV was turned on by PC. AppleTV was turned on by TV. For 1 sec TV was on hdmi 1 while Apple TV is booting up and then it's taking over the source. 


[log.txt](https://github.com/user-attachments/files/16875470/log.txt)

**Comment by JPersson77** [comment]
But then it sounds like a CEC issue. The app managed to initially and correctly switch to hdmi1 in step 4 if I understand correctly? But then apple tv forced another switch immediately after.

**Comment by vladimir-fsh** [comment]
@JPersson77 exactly https://www.reddit.com/r/LGOLED/comments/12j353m/how_to_stop_lg_switching_to_appletv_input_on/

So it's not the issue of your app, thank you anyway. It's small inconvenience for me, the rest functionality is very useful

**Comment by JPersson77** [comment]
Yeah that link potentially explains it well.

The app cannot prevent CEC communications between other devices of course. Potentially the hdmi switching command can be delayed or sent multipel times to "override"

**Comment by vladimir-fsh** [comment]
@JPersson77 I also thought about the similar fix. Well, will see, might be this issue is affecting someone else and then it would raise the priority

**Comment by hapklaar** [comment]
> Hi @hapklaar there are no issues with your log, everything looks as expected. Since the TV powers on fine and responds to the request to switch HDMI, but doesn't actually do it, Isuspect that the TV needs slightly more time after being powered on before the HDMI-switch command is sent. Currently on vacation so it'll take a few weeks but I'll have a look

@JPersson77 did you find some time to have a look? Maybe a workaround in the form of a configurable delay could help.

**Comment by JPersson77** [comment]
Hi there been a little busy nut this is not forgotten about

**Comment by Sindorius** [comment]
Same issue here, LG C3. It used to work normally but recently I took my HDMI cable out and put it back in and my TV switched from my PC being HDMI 3 to HDMI 1 and when I set and applied LGTV Companion to use HDMI 1 instead it still tries to set it to 3 on wakeup.

**Comment by JPersson77** [comment]
hi @Sindorius this is a similar but not the same issue. 

From your description It sounds like the configuration was not saved properly. Can you confirm you pressed "apply" in the main gui before exiting the GUI? Do you mind enabling debug level logging and upload a log to pastebin and I will have a look

**Comment by Sindorius** [comment]
@JPersson77 Yeah I pressed apply. I even tried setting it to another HDMI input #, hitting apply, and setting it back to 1, and hit apply again to see if I could resave settings. But same issue.

How do I enable debug level logging and see the log?

**Comment by Sindorius** [comment]
@JPersson77 Yeah I pressed apply. I even tried setting it to another HDMI input #, hitting apply, and setting it back to 1, and hit apply again to see if I could resave settings. But same issue.

Here's the log output: https://pastebin.com/ZdCBAd9h

**Comment by JPersson77** [comment]
From the log supplied the app is setting the TV to hdmi input 1 so that, seems correct. Is something else setting it to hdmi input3

**Comment by Sindorius** [comment]
Not that I'm aware of. I checked the TV settings and there was something to have it go to 'recent input' when turned back on but that's 1 too.

**Comment by JPersson77** [comment]
Hi @hapklaar @vladimir-fsh @Sindorius please see if the latest version v4.1.0 resolves some or all of your issues

**Comment by hapklaar** [comment]
Unfortunately my laptop now blocks this executable (Windows Defender ASR, managed from Intune by my employer), so not able to test at the moment unfortunately...

**Comment by vladimir-fsh** [comment]
> Hi @hapklaar @vladimir-fsh @Sindorius please see if the latest version v4.1.0 resolves some or all of your issues

Thanks for the fix. I can check it after gpu upgrade, in January. Now my pc is without gpu

**Comment by hapklaar** [comment]
@JPersson77 I bribed my admin to allow me to run it on my laptop again, but unfortunately it didn't fix it for me. I set the delay to 3 seconds, but don't see any change in behavior. 

It turns on my TV, but doesn't change the input to HDMI2 where my laptop is connected.

new log: https://pastebin.com/r8NXRNTB

PS it all works fine on my other PC also connected to the same TV on HDMI1, so it looks like it has something to do with the laptop. Can't imagine what it could be though, as it's a simple web request that shouldn't be any different.

**Comment by JPersson77** [comment]
Hi @hapklaar i can see that it seems  the app correctly communicates with the TV. Can it be that you expect the power on/off test function to also set the hdmi channel (it doesn't)?

**Comment by hapklaar** [comment]
Ooh ok yes I expected the power tests to do the same as in actual use, sorry for that.

However now also tested using the following steps:

1. Close laptop lid ==> C2 turns off correctly
2. Turn on TV, set it to HDMI1, turn off TV
3. Open laptop lid ==> C2 turns on correctly, but remains on input HDMI1

new part of the log here: https://pastebin.com/tcEuakDM

**Comment by JPersson77** [comment]
The last log shows that WOL packets was sent (which wake the TV) but does not show a full connection where the app is communicating with the TV.  Which in itself explains why the HDMI input was not set. But additionally I think it also points at something which needs fixing, because the app is retrying the connection too frequently and times out too quickly. I don't think I've seen a situation where the webos handshake was sent, immediately followed by a reset connection. This may need to be adjusted for in the code

**Comment by hapklaar** [comment]
Got it. If there are any other tests you'd like me to perform, let me know.

**Comment by JPersson77** [comment]
@hapklaar PM me on discord if you would like a test version

https://discord.gg/7KkTPrP3fq

**Comment by JPersson77** [comment]
I think v4.3.0 should clear any remaining issues with the HDMI-switching so I'm closing the issue. Feel free to reopen as needed

**Comment by vladimir-fsh** [comment]
@JPersson77 I've tested new version and it's working for me with 7 second delay. Thank you so much!
